container.go

```diff
@@ -410,6 +448,8 @@ func (c *Container) start() error {
 		return err
 	}
 
+	c.storeMounts()
```

Comment:
I am not saying this is not a good idea to do that at the pod/container level, but according to the way we modify this list of mounts from inside the agent, the agent itself should store the new list. I would be more than happy to see this modification (list of mounts I mean) out of the agent implementation but I think it is a bit of work to refactor this.
Therefore, according to the way it is implemented here (modified inside the agent implementation), I think the agent implementation should take care of storing this mount list.